server/webapp/WEB-INF/rails.new/app/views/agents/_agents_table.html.erb

```diff
@@ -10,16 +10,32 @@
 Util.on_load(function() {
     jQuery("#select_all_agents").live('click',function() {
         var val = this.checked;
-        jQuery(".agent_select").each(function() {
-            this.checked = val;
-        });
+        jQuery(".agent_select").prop("checked", val);
     });
 
-    jQuery("input.agent_select").live("click", function() {
-        var unchecked_check_boxes = jQuery("input.agent_select:not(:checked)");
-        if(unchecked_check_boxes.length > 0) {
-            jQuery("#select_all_agents").removeAttr("checked");
-        }
+    var lastChecked = null;
+
+    jQuery(document).ready(function() {
+        jQuery('.agent_select').live('click', function(e) {
+            if (!this.checked) {
```

Comment:
The problem with ```live``` is it checks your dom for current and future elements which ends up making this script slow. 
Should take this for a spin with a lot more agents.